rollup/internal/orm/pending_transaction.go

```diff
@@ -85,6 +85,19 @@ func (o *PendingTransaction) GetPendingOrReplacedTransactionsBySenderType(ctx co
 	return transactions, nil
 }
 
+// GetCountPendingTransactionsBySenderType retrieves number of pending transactions filtered by sender type
+func (o *PendingTransaction) GetCountPendingTransactionsBySenderType(ctx context.Context, senderType types.SenderType) (int64, error) {
+	var count int64
+	db := o.db.WithContext(ctx)
+	db = db.Model(&PendingTransaction{})
+	db = db.Where("sender_type = ?", senderType)
```

Comment:
pls add a new index for this query, current indices might not help:
```
CREATE UNIQUE INDEX unique_idx_pending_transaction_on_hash ON pending_transaction(hash);
CREATE INDEX idx_pending_transaction_on_sender_type_status_nonce_gas_fee_cap ON pending_transaction (sender_type, status, nonce, gas_fee_cap);
CREATE INDEX idx_pending_transaction_on_sender_address_nonce ON pending_transaction(sender_address, nonce);
```

Reply:
I think the second index `ON pending_transaction (sender_type, status, nonce, gas_fee_cap);` should be helpful because the first 2 columns are the same as in the query `sender_type, status`. Also, checked it with postrgres doc, 

> A multicolumn B-tree index can be used with query conditions that involve any subset of the index's columns, but the index is most efficient when there are constraints on the leading (leftmost) columns.

But if you still unsure I can add it

Reply:
You can try generating some sample queries, and using `explain analyze` to see if it can use some indices. if so, we don't need to add a new one for this query.

Reply:
`explain analyze select count(*) from pending_transaction where sender_type = 1 and status = 1;` -> 
```
 Aggregate  (cost=8.45..8.46 rows=1 width=8) (actual time=2.604..2.604 rows=1 loops=1)
   ->  Index Only Scan using idx_pending_transaction_on_sender_type_status_nonce_gas_fee_cap on pending_transaction  (cost=0.42..8.45 rows=1 width=0) (actual time=2.596..2.597 rows=1 loops=1)
         Index Cond: ((sender_type = 1) AND (status = 1))
         Heap Fetches: 1375
 Planning Time: 0.093 ms
 Execution Time: 2.638 ms
```

Execution time is very fast, much faster than querying for example `sender_type` and `nonce` . So I think it's ok to leave existing indexes

Reply:
cool. thanks for testing.